src/System.Data.SqlClient/src/System/Data/SqlClient/SNI/SNIPacket.cs

```diff
@@ -259,48 +238,10 @@ public void ReadFromStream(Stream stream)
         /// <param name="stream">Stream to write to</param>
         public void WriteToStream(Stream stream)
         {
-            stream.Write(_data, 0, _length);
-        }
-
-        /// <summary>
-        /// Get hash code
-        /// </summary>
-        /// <returns>Hash code</returns>
-        public override int GetHashCode()
-        {
-            return base.GetHashCode();
+            stream.Write(_data, _header, _length);
         }
 
-        /// <summary>
-        /// Check packet equality
-        /// </summary>
-        /// <param name="obj"></param>
-        /// <returns>true if equal</returns>
-        public override bool Equals(object obj)
```

Comment:
What's the purpose to remove these public functions?

Reply:
Packets are never compared to each other. If in further work they are compared it will be in a pool and the reference equality will be all that is needed. As such these methods aren't needed or used currently.

Reply:
I suggest to remain what we have before even if it's not used currently.
There should be a reason why they were added.

Reply:
They were originally added with the intention of using them when pooling managed packets. The packet pooling was never fully written and integrated so they weren't used. See the removed `WritePacketCache` code in `TdsParserStateObjectManaged`. Are you sure you want dead and trivial code keeping?